modules/terminate-agent-hook/lambda/lambda_function.py

```diff
@@ -102,19 +163,19 @@ def handler(event, context):
         }))
         try:
             client.terminate_instances(InstanceIds=_terminate_list, DryRun=False)
-            return f"Terminated instances {', '.join(_terminate_list)}"
```

Comment:
Is this removing the log message that confirms that we actually terminated the instances successfully?

Also, would it make sense to split the instance termination out into a separate function too so that the handler is just:

```python
terminate_running_instances(...)
cancel_active_spot_requests(...)
```

Reply:
Guess I forgot to replace this with a Json log. Thanks!

Reply:
I think we shouldn't create a second lambda as we have to make sure that everything is killed. In case of two functions it could happen, that we kill the EC2s but new spot instances are created because the second lambda function has not been started yet.

I changed the code to terminate the spor requests first and then killing the EC2 instances. This way we should be sure that there is nothing running.